### PR TITLE
NW 4 - Maha Malik - JS Core 1 - Week 2

### PR DESCRIPTION
**Pull request by maham511**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Maha Malik
- Your City: NW4
- Your Slack Name: Maha Malik

# Homework Details

- Module: JS Core 1
- Week: 2

# Notes

- What did you find easy?
Basic concepts are solidifying, finding easier how to approach problems using if statements/loops etc

- What did you find hard?
Storing functions, and also calling them in variables. Especially when nesting functions/variables etc

- What do you still not understand?
Same as above

- Any other notes?

### CODE REVIEW

File: mandatory/2-function-creation.js [hunk 1]
```diff
@@ -25,13 +25,20 @@ Write a function that:
...
 function formatPercentage(num) {
-  const strPercent = num + '%';
-  return num.toFixed(2);
-  const keep100 =  num > 100? '100%': strPercent;
-  // return keep100.toFixed(2); 
-  
+  const twoDecimal = Math.round(num * 100) / 100;  // Rounds off num to upto 2 decimals if floats, leaves integers as is
+  const strPercent = twoDecimal + "%";              // twoDecimal changes into string value after concatenate with '%'
+  const keep100 = num >= 100 ? "100%" : strPercent;  // If num >= 100, output will say '100%'; else outputs expected strPercent values (i.e formatted and rounded)
+
```
Comment (omerffaruk) [inline mandatory/2-function-creation.js:31]: Excellent job Maha, really like how clean the code is. I also see you use ternary operator at line 30 :) Brilliant

**Review by omerffaruk — APPROVED** [review]
Excellent job

File: mandatory/2-function-creation.js [hunk 2]
```diff
@@ -50,25 +50,13 @@ function tidyUpStrings(arrayOfStrings) {
...
-
-
-
-//  //Trim
-//   arrayOfStrings.map((strArr) => strArr.trim());
-//   console.log(arrayOfStrings());
+ const smlCap = strArr.map((word) => word.toLowerCase()); // .map() iterates through all items and turns into lowercase
+ const trimName = smlCap.map((word) => word.trim());     //  Iterates smlCap, & trims spaces with .trim() 
+ const noSlash = trimName.map((word) => word.replace('/', ''));  // Iterates through trimName & removes forward slash using replace .map() and .replace() methods
+
```
Comment (omerffaruk) [inline mandatory/2-function-creation.js:56]: Great use of map method 👍 very good explanations

**Review by omerffaruk — APPROVED** [review]

**Comment by omerffaruk** [comment]
Good variable names with camelCase style, nice indentation, easy to understand clean code, clever solutions, excellent job Maha 💯

File: mandatory/2-function-creation.js (hunk 1 above)
Comment (Maziarmajd) [reply mandatory/2-function-creation.js:31]: Great job Maha. clean code and brilliant Idea

File: mandatory/2-function-creation.js (hunk 2 above)
Comment (Maziarmajd) [reply mandatory/2-function-creation.js:56]: very complete explanation. great job Maha

**Comment by maham511** [comment]
Thanks so much Omer and Maziar for such positive reviews! :)

Regards,
Maha

File: mandatory/2-function-creation.js (hunk 2 above)
Comment (maham511) [reply mandatory/2-function-creation.js:56]: Thanks Omer and Maziar!  :)

**Pull request closed without merging**